src/EPR.CommonDataService.Core/Services/SubmissionsService.cs

```diff
@@ -58,18 +58,30 @@ public async Task<IList<ApprovedSubmissionEntity>> GetApprovedSubmissionsWithAgg
     {
         logger.LogInformation("{LogPrefix}: SubmissionsService - GetApprovedSubmissionsWithAggregatedPomData: Get approved submissions after {ApprovedAfter}, for periods {Periods}", _logPrefix, approvedAfter.ToString(CultureInfo.InvariantCulture), periods);
 
-        var sql = "EXECUTE rpd.sp_GetApprovedSubmissionsWithAggregatedPomDataIncludingPartialV3 @ApprovedAfter, @Periods";
+        var sql = "EXECUTE rpd.sp_GetApprovedSubmissionsWithAggregatedPomDataForDirectRegistrantsAndComplianceSchemeV1 @ApprovedAfter, @Periods";
         logger.LogInformation("{LogPrefix}: SubmissionsService - GetApprovedSubmissionsWithAggregatedPomData: executing query {Sql}", _logPrefix, sql);
 
         try
         {
             databaseTimeoutService.SetCommandTimeout(accountsDbContext, 120);
-            var paginatedResponse = await accountsDbContext.RunSqlAsync<ApprovedSubmissionEntity>(sql,
+
+            var result = await accountsDbContext.RunSqlAsync<ApprovedSubmissionEntity>(
+                sql,
                 new SqlParameter("@ApprovedAfter", SqlDbType.DateTime2) { Value = approvedAfter },
-                new SqlParameter("@Periods", SqlDbType.VarChar) { Value = periods ?? (object)DBNull.Value });
+                new SqlParameter("@Periods", SqlDbType.VarChar) { Value = periods ?? (object)DBNull.Value }
+            );
+
+            if (result == null || !result.Any())
+            {
+                logger.LogWarning("{LogPrefix}: SubmissionsService - GetApprovedSubmissionsWithAggregatedPomData: No approved submissions found for the specified criteria.",
+                    _logPrefix);
+                return new List<ApprovedSubmissionEntity>();
```

Comment:
Is the consumer expecting / checking for empty array instead of null ?

Reply:
checks for empty array so its fine